File: Chapter07/C#/Project/ASPAJAX/4-AdoDataService/AdoDataService/SilverlightResourceDataService.svc.cs
﻿using System;
using System.Collections.Generic;
using System.Data.Services;
using System.Linq;
using System.ServiceModel.Web;
using System.Web;

namespace AdoDataService
{
   public class SilverlightResourceDataService : DataService<SilverlightResourceEntities>
    {
        // This method is called only once to initialize service-wide policies.
        public static void InitializeService(IDataServiceConfiguration config)
        {
            // TODO: set rules to indicate which entity sets and service operations are visible, updatable, etc.
            // Examples:
             config.SetEntitySetAccessRule("Resource", EntitySetRights.All);
             config.SetEntitySetAccessRule("ResourceType", EntitySetRights.All);
            // config.SetServiceOperationAccessRule("MyServiceOperation", ServiceOperationRights.All);
        }
    }
}

File: Chapter07/VB/VBProject_07/ASPAJAX/4-AdoDataService/AdoDataService/SilverlightResourceDataService.svc.vb
Imports System
Imports System.Collections.Generic
Imports System.Data.Services
Imports System.Linq
Imports System.ServiceModel.Web
Imports System.Web

' Namespace AdoDataService

    Public Class SilverlightResourceDataService
        Inherits DataService(Of SilverlightResourceEntities)
        '  This method is called only once to initialize service-wide policies.
    Public Shared Sub InitializeService(config As IDataServiceConfiguration)

        '  TODO: set rules to indicate which entity sets and service operations are visible, updatable, etc.
        '  Examples:
        config.SetEntitySetAccessRule("Resource", EntitySetRights.All)
        config.SetEntitySetAccessRule("ResourceType", EntitySetRights.All)
        '  config.SetServiceOperationAccessRule("MyServiceOperation", ServiceOperationRights.All)
    End Sub '   InitializeService
End Class   '   SilverlightResourceDataService
' End Namespace   '   AdoDataService
' ..\Project_07\ASPAJAX\4-AdoDataService\AdoDataService\SilverlightResourceDataService.svc.cs
